### PR TITLE
Add background squares and refactor coloring system

### PR DESCRIPTION
**Pull request by HelloChrisLowe**

### CODE REVIEW

File: src/chess_board.rs [hunk 1]
```diff
@@ -180,50 +184,60 @@ impl ChessBoard
                 let piece = BitboardPiece::from_file_rank(x, 8 - y - 1);
                 if self.allies.contains(piece)
                 {
-                    print!("\x1b[1;32m");
+                    print!("{}", color::gen_color_escape_code(color::FG_GREEN, is_white_space));
```
Comment (Isaac-Lozano) [inline src/chess_board.rs]: Need to decouple color functionality and board state. So remove the bool parameter to `color::gen_color_escape_code()` and add a background color as a second parameter.

File: src/color.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+pub const FG_WHITE: u32     = 37;
+
+pub const BG_BLACK: u32     = 40;
+pub const BG_RED: u32       = 41;
+pub const BG_GREEN: u32     = 42;
+pub const BG_YELLOW: u32    = 43;
+pub const BG_BLUE: u32      = 44;
+pub const BG_MAGENTA: u32   = 45;
+pub const BG_CYAN: u32      = 46;
+pub const BG_WHITE: u32     = 47;
```
Comment (Isaac-Lozano) [inline src/color.rs]: Make a Color enum and add these as elements, then add a `as_background_color(self) -> u32` and `as_foreground_color(self) -> u32` functions.

Comment (Isaac-Lozano) [reply src/color.rs]: We only need the colors themselves, and not if they're bg/fg, so something like:
```rust
pub enum Color
{
    Black,
    Red,
    Green,
    Yellow,
    Blue,
    Magenta,
    Cyan,
    White,
}
```

File: src/color.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+pub fn gen_color_escape_code(fgcolor: u32, is_white_space: bool) -> String
```
Comment (Isaac-Lozano) [inline src/color.rs]: As mentioned before, change the function to:
```rust
pub fn gen_color_escape_code(fg: Color, bg: Color) -> String
{
    ...
}
```

File: src/chess_board.rs [hunk 4]
```diff
@@ -180,50 +184,60 @@ impl ChessBoard
...
-                    print!("P ");
+                    print!(" P ");
                 }
                 else
                 {
-                    print!("- ");
+                    print!("   ");
                 }
 
                 print!("\x1b[0m");
```
Comment (Isaac-Lozano) [inline src/chess_board.rs]: Perhaps add a `gen_reset_color()` or something similar to color.rs to replace this line.

**Review by Isaac-Lozano — CHANGES_REQUESTED** [review]
Still need to restructure things to make it more Rusty, but looks good so far.

**Comment by HelloChrisLowe** [comment]
Ready for re-review

File: src/color.rs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    assert_eq!(37, get_foreground_color_code(Color::White));
+}
+
+#[test]
+fn test_get_background_color_code()
+{
+    assert_eq!(40, get_background_color_code(Color::Black));
+    assert_eq!(43, get_background_color_code(Color::Yellow));
+    assert_eq!(47, get_background_color_code(Color::White));
+}
```
Comment (Isaac-Lozano) [inline src/color.rs:67]: Tests should separated from other code by putting it in a module. Similar to tests in the minimax crate/repo.

**Review by Isaac-Lozano — APPROVED** [review]
It's all good except for tests being outside of a module. Also a bit of extra newlines after the end of the file but that's fine. You can submit another pull request to fix these if you want, but this will be accepted.

**Pull request merged**